x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/constants/data_types_definition.tsx

```diff
@@ -122,15 +122,43 @@ export const TYPE_DEFINITION: { [key in DataType]: DataTypeDefinition } = {
     documentation: {
       main: '/date.html',
     },
+    description: () => (
+      <p>
+        <FormattedMessage
+          id="xpack.idxMgmt.mappingsEditor.dataType.dateLongDescription"
+          defaultMessage='Date fields accept strings containing formatted dates, e.g. "2015-01-01" or "2015/01/01 12:10:30", a long number representing milliseconds-since-the-epoch, or an integer representing seconds-since-the-epoch. Multiple date formats can be specified. Dates with timezone information will be converted to UTC internally.'
+        />
+      </p>
+    ),
   },
   date_nanos: {
     label: i18n.translate('xpack.idxMgmt.mappingsEditor.dataType.dateNanosDescription', {
-      defaultMessage: 'Date nanos',
+      defaultMessage: 'Date nanoseconds',
     }),
     value: 'date_nanos',
     documentation: {
       main: '/date_nanos.html',
     },
+    description: () => (
+      <p>
+        <FormattedMessage
+          id="xpack.idxMgmt.mappingsEditor.dataType.dateNanosLongDescription"
+          defaultMessage="Date nanoseconds fields store dates in nanosecond resolution, which limits its range of dates. Aggregations are still on the millisecond resolution. If you need to store dates in millisecond resolution, use the {date}."
```

Comment:
I wondered what the range of dates was limited to and why they were limited, until I read the docs and learned its limited to the range from 1970 to 2262 because dates will be stored as a long value representing epoch time in ns.

I think we should either remove "which limits its range of dates" or, if you think it's important to include, add the range to which it's limited (1970-2262). I think either choice would provide a more complete picture to the user.

Reply:
Good point. I removed "which limits its range of dates".

Reply:
Date nanoseconds fields store dates in nanosecond resolution. Aggregations remain in millisecond resolution.  To store dates in millisecond resolution, use the date data type.

Reply:
Latest copy:
<img width="728" alt="Screen Shot 2020-01-08 at 9 44 21 AM" src="https://user-images.githubusercontent.com/5226211/71987411-2033b280-31fc-11ea-8ed1-8add8637113f.png">